### PR TITLE
Generate mmm-modes automatically based on a list

### PR DESCRIPTION
**Pull request by crshd**
This "fixes"  #6571 

**Question:**
Should we really add all possible combinations of languages, or let the user (Spacemacsian) take care of that themselves?

If we keep adding definitions to the core, we'll end up with a terribly long list...

### CODE REVIEW

**Comment by wookietreiber** [comment]
Shouldn't there be a more DRY why of supporting **all** languages?

**Comment by crshd** [comment]
I don't think there's a fully automated way. But maybe something like [this](http://jblevins.org/log/mmm)?

**Comment by wookietreiber** [comment]
Yes, this looks definitely less boilerplaty.

File: layers/+lang/markdown/packages.el [hunk 1]
```diff
@@ -42,6 +42,18 @@
 (defun markdown/post-init-smartparens ()
   (add-hook 'markdown-mode-hook 'smartparens-mode))
 
+;; from Jason Blevins http://jblevins.org/log/mmm
+(defun markdown/mmm-auto-class (lang &optional submode)
```
Comment (crshd) [inline layers/+lang/markdown/packages.el]: This should be `spacemacs//markdown-mmm-auto-class`, shouldn't it?

**Comment by wookietreiber** [comment]
I would also like a small section in `layers/+lang/markdown/README.org` detailing how and where to add these settings to the `.spacemacs` file.

**Comment by crshd** [comment]
[There is](https://github.com/crshd/spacemacs/tree/develop/layers/%2Blang/markdown#automatic-mmm-mode-generation).
I guess I could add another section for the not-directly relating modes. But I'm not yet sure wether I want to change how those are handled.

**Comment by wookietreiber** [comment]
My bad, was looking at master branch. LGTM.

**Comment by TheBB** [comment]
That's not ugly!

**Comment by syl20bnr** [comment]
Thank you ! 💜 
Indeed this is not ugly at all!
I made minor cleanups in commit: https://github.com/syl20bnr/spacemacs/commit/4fd895a9
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**